### ISSUE TITLE
Density [Performance suite] should allow starting 30 pods per node (Failed 13 times in the last 30 runs. Stability: 56 %)

### ISSUE THREAD

**Issue by quinton-hoole** [kubernetes/kubernetes#9485]

**Comment by quinton-hoole** [comment]
@wojtek-t @fgrzadkowski @davidopp FYI.  This might be a dupe of another issue - just making sure that all of our currently flaky e2e tests are covered by issues.

**Comment by quinton-hoole** [comment]
Master tracking bug: #9312

**Comment by fgrzadkowski** [comment]
This for regular gce tests, right? For scalability jenkins job this is very stable (close to 100%).

**Comment by quinton-hoole** [comment]
Yes, regular e2e-gce tests.

**Comment by quinton-hoole** [comment]
It's in the e2e-gce-flaky run at the moment.  It's entirely possible that some other test in that run is making this one fail.

**Comment by quinton-hoole** [comment]
The error is that the number of high latency requests (currently defined to be >3 sec) is greater than zero (typically around 12).
Perhaps we can relax that threshold in the test?  Increase 3 sec to more?  Do we understand why they're taking so long?

https://github.com/GoogleCloudPlatform/kubernetes/blob/master/test/e2e/density.go#L92

**Comment by fgrzadkowski** [comment]
My guess is that the metrics are too high because of some previous/other tests. Currently we do not have mechanism to compute delta of the latency metrics (though it should not be very hard to implement).

@quinton-hoole Do the tests run in parallel?

**Comment by quinton-hoole** [comment]
@fgrzadkowski No, in this case they're running in serial.

**Comment by fgrzadkowski** [comment]
This test fails because latency metrics are too high. I looked into this and my current theory is that the flaky runs are due to previous tests that affected latency (we do not reset metrics between tests). I'll send a PR to verify this.

I see two solutions:
1. Measure latency by reading full histogram before and after the test and manually compute percentiles.
2. Disable this tests in regular jenkins job and rely on the one issued in scalability job.

I think that fixing 1) is not necessary at this point. I'd suggest doing 2).

@quinton-hoole WDYT?

**Comment by fgrzadkowski** [comment]
After #9574 is checked few runs in jenkins and in all cases it caused by some random calls from other cases (e.g. POSTing limit ranges). I believe that we should just rely on density in scalability job.

@quinton-hoole Thoughts?

**Comment by wojtek-t** [comment]
+1 - I definitely agree that since we have a pretty stable scalability suite on Jenkins, we can drop the density test from regular GCE.

On the other hand, we shouldn't have requests with >3s latency in 2-node clusters - @fgrzadkowski did you look into it?

**Comment by fgrzadkowski** [comment]
@wojtek-t Not very deeply. I've noticed that sometimes this happens even during cluster intialization.

**Comment by quinton-hoole** [comment]
@wojtek-t @fgrzadkowski 

Two comments:

1. Having "hermetic" tests, which are independent of other tests, is our goal for e2e. To this end, measuring the metrics for only this test, and not previous ones would be a good improvement to the test. post-v1.0.

2. I share @wojtek-t 's concerns as to why exactly we're seeing such high latency calls in small clusters.  It would be good to get to the bottom of that one, independent of this test, and of v1.0.

But, for 1.0, lets rely on the this test running in relative isolation in our scalability job.  
Moving to post 1.0.

**Comment by quinton-hoole** [comment]
Oh, and I'll turn off this test in the general e2e-gce continuous integration test.

**Comment by alex-mohr** [comment]
kubernetes-e2e-gce-scalability is now in the critical builds set that blocks submit queue, so re-opening given failure rates recently.

Node odd output of [Skipped] yet build marked a failure -- I'm missing some subtlety re: skipped.

http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-scalability/2921/
Density [Skipped] [Performance suite] should allow starting 30 pods per node
/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/density.go:111
There should be no high-latency requests
Expected
    <int>: 1
not to be >
    <int>: 0

Other occurrences:
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-scalability/2906/
(which also timed out)

And similar but different issue?
http://kubekins.dls.corp.google.com/view/Critical%20Builds/job/kubernetes-e2e-gce-scalability/2871/

Density [Skipped] [Performance suite] should allow starting 30 pods per node

/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/test/e2e/density.go:398
Expected error:
    <*errors.errorString | 0xc20cafcec0>: {
        s: "too high pod startup latency 99th percentil: 6.703608835s",
    }
    too high pod startup latency 99th percentil: 6.703608835s
not to have occurred

**Comment by wojtek-t** [comment]
Kubekins 2871 run was fixed by #16428

**Comment by wojtek-t** [comment]
Kubekins 2906 was something strange - metrics were 10x larger than in all other tests - I'm pretty sure this was related to some GCE issues.
